comp/dogstatsd/server/parse_events.go

```diff
@@ -169,10 +167,7 @@ func (p *parser) applyEventOptionalField(event dogstatsdEvent, optionalField []b
 	case bytes.HasPrefix(optionalField, eventTagsPrefix):
 		newEvent.tags = p.parseTags(optionalField[len(eventTagsPrefix):])
 	case p.dsdOriginEnabled && bytes.HasPrefix(optionalField, localDataPrefix):
-		newEvent.localData, err = p.resolveLocalData(optionalField[len(localDataPrefix):])
-		if err == nil {
-			newEvent.containerID = newEvent.localData.ContainerID
-		}
+		newEvent.localData, err = origindetection.ParseLocalData(string(optionalField[len(localDataPrefix):]))
```

Comment:
Same as with previous PR, we probably don't want to discard event completely if parsing fails.

The error message should probably be decorated with the description which field has failed to parse (same as what we do in metric sample).

```suggestion
		newEvent.localData = parseLocalData(optionalField)
```

And use the helper in all three cases in metrics, events and service checks:

```
func parseLocalData(field []byte) string {
  val, err := origindetection.ParseLocalData(string(field[len(localDataPrefix):]))
  if err != nil {
    log.Errorf("failed to parse c: field %q: %v", field, err)
    return ""
  }
  return val
}
```

Reply:
Done here: https://github.com/DataDog/datadog-agent/pull/32839/commits/b4c27ad497a0cd661e42db575ae4e0f049b4bdbe